hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -258,18 +258,19 @@ public class HoodieWriteConfig extends HoodieConfig {
           "**Note** This is being actively worked on. Please use "
               + "`hoodie.datasource.write.keygenerator.class` instead.");
 
-  public static final ConfigProperty<Boolean> COMPLEX_KEYGEN_OLD_ENCODING = ConfigProperty
-      .key("hoodie.write.complex.keygen.old.encoding")
-      .defaultValue(true)
+  // keep both configs for table version 8 and below
+  // change naming to be new and then flip default to false
+  public static final ConfigProperty<Boolean> COMPLEX_KEYGEN_NEW_ENCODING = ConfigProperty
+      .key("hoodie.write.complex.keygen.new.encoding")
+      .defaultValue(false)
       .markAdvanced()
-      .sinceVersion("1.1.0")
```

Comment:
Taking back my previous comment, I think we still need to mark this with `.sinceVersion("1.1.0")` since it is used for writing table version 8 and below.